src/android/KnoxPluginEnabled.kt

```diff
@@ -1,109 +1,170 @@
+        super.onDestroy()
+    }
+
+    override fun execute(
+        action: String,
+        args: JSONArray,
+        callbackContext: CallbackContext
+    ): Boolean {
+        Timber.v("execute action '$action'")
+        when (action) {
+            ACTION_IS_ENABLED -> {
+                Timber.v("responding with 'true'")
+                callbackContext.success(JSONObject().put("enabled", KNOX_ENABLED))
+            }
 
             ACTION_GET_VERSION_INFO -> {
                 cordova.threadPool.execute {
-                    try {
-                        val context = cordova.context
-                        val edm = EnterpriseDeviceManager.getInstance(context)
-                        val applicationPolicy = edm.applicationPolicy
-                        val knoxAppVersion = applicationPolicy.getApplicationVersion(context.packageName)
-                        val result = JSONObject()
-                            .put(KEY_KNOX_APP_VERSION, knoxAppVersion)
-                        callbackContext.success(result)
-                    } catch (ex: Exception) {
-                        val errorMessage = "Failed to fetch version info: ${ex.message}"
-                        Timber.e(errorMessage, ex)
-                        callbackContext.error(errorMessage)
-                    }
+                    getVersionInfo(callbackContext)
+                }
+            }
+
+            ACTION_SHUTDOWN -> {
+                cordova.threadPool.execute {
+                    shutdown(callbackContext)
+                }
+            }
+
+            ACTION_REBOOT -> {
+                cordova.threadPool.execute {
+                    reboot(callbackContext)
                 }
             }
 
-			else -> {
-				Timber.w("rejecting unsupported action '$action'")
-				callbackContext.error("Action $action is not implemented in KnoxPlugin.")
-				return false
-			}
-		}
-		return true
-	}
+            else -> {
+                Timber.w("rejecting unsupported action '$action'")
+                callbackContext.error("Action $action is not implemented in KnoxPlugin.")
+                return false
+            }
+        }
+
+        return true
+    }
+
+    private fun getVersionInfo(callbackContext: CallbackContext) {
+        try {
+            val context = cordova.context
+            val edm = EnterpriseDeviceManager.getInstance(context)
+            val applicationPolicy = edm.applicationPolicy
+            val knoxAppVersion =
+                applicationPolicy.getApplicationVersion(context.packageName)
+            val result = JSONObject()
+                .put(KEY_KNOX_APP_VERSION, knoxAppVersion)
+            callbackContext.success(result)
+        } catch (ex: Exception) {
+            val errorMessage = "Failed to fetch version info: ${ex.message}"
+            Timber.e(ex, errorMessage)
+            callbackContext.error(errorMessage)
+        }
+    }
+
+    private fun shutdown(callbackContext: CallbackContext) {
```

Comment:
Still not sure if they want this. We don't do anything with it currently in PCKS. I've written in the shutdown ticket that we need to confirm with Mariam. Will keep you posted.

Reply:
Is there any harm in including the API and just not using it in monorepo? Right now this wont do anything unless we explicitly tell it to